### ISSUE TITLE
Algo falls more frequently in local minima (let's try with a issue in this repo)

### ISSUE THREAD

**Issue by mallibus** [nickprock/pynuTS#8]
Hi Nicola
I ran some experiments with the updated clustering library, with the introduction of the stop criteria.
I noticed that the algo falls more frequently in local minima resulting in wrong clustering. 
See attachment 
![bad output example](https://user-images.githubusercontent.com/45511108/109402233-6a25ad80-7954-11eb-9ceb-0d7c6969f2d2.png)

I think this is not a real bug but something to manage, e.g. using the parameter patience, still I have in mind 2 features to be added to the clustering lib in order to do that:

1) introduce a random seed parameter in the DTWKmeans class or in the fit method - this helps in making reproduceable runs useful for unit testing and for troubleshooting, e.g. to nail down the cases of algo stuck in local minima

2) introduce a new parameter [similar to the sklearn n_init in KMeans](https://scikit-learn.org/stable/modules/generated/sklearn.cluster.KMeans.html) in the DTWKmeans

I have 1) ready, 2 yet to be done.
As those changes would impact the core API of the library I'm not starting the integration before an OK from you

**Comment by nickprock** [comment]
Hello @mallibus , you can start the integration.
